mitxgraders/baseclasses.py

```diff
@@ -40,15 +47,63 @@ def validate_config(self, config):
         """
         return voluptuous_validate(config, self.schema_config)
 
+    @classmethod
+    def register_defaults(cls, values_dict):
+        """Saves a dictionary of values to override the defaults"""
+        if cls.default_values is None:
+            cls.default_values = {}
+        cls.default_values.update(values_dict)
+
+    @classmethod
+    def clear_registered_defaults(cls):
+        """Clears all registered defaults"""
+        cls.default_values = None
+
     def __init__(self, config=None, **kwargs):
         """
         Validate the supplied config for the object, using either a config dict or kwargs,
-        but not both.
+        but not both. Also loads any registered defaults.
         """
+        # Select the appropriate source for configuration
         if config is None:
-            self.config = self.validate_config(kwargs)
+            use_config = kwargs
         else:
-            self.config = self.validate_config(config)
+            use_config = config
+
+        # Apply any registered defaults before overriding with the given configuration
+        if isinstance(use_config, dict):
+            use_config = self.apply_registered_defaults(use_config)
+
+        # Validate the configuration
+        self.config = self.validate_config(use_config)
+
+    def apply_registered_defaults(self, config):
```

Comment:
Hmm. It's not quite that simple, because this is called during `__init__`, and the log is constructed during `__call__`... I think I can get around it by making a copy of the dict though.

Reply:
Or have a `self.config_debug_log` created during init, and concatenate them during call?

Reply:
I've partially implemented this. I'm going to need the changes I made to logging in #252 to finish it up though. I'll make a separate PR to do this once this PR and that one are merged.

Reply:
(At least, to avoid the dreaded merge-conflicts...)